app/queries/homepage/articles_query.rb

```diff
@@ -67,6 +67,7 @@ def call
 
     def filter
       @relation = @relation.full_posts
+      @relation = @relation.where("title ILIKE ?", "%#{@search_fields}%") if @search_fields.present?
```

Comment:
I am not sure of the complete functionality here but Ilike query on title will be slower.
We should add an index on title column. Also normal index won't help here. We should use something pg_trgm index, used for other column here.
Maintainer can tell better 
@dpgaharwal

Reply:
Thank you for the feedback! I’ll address the performance concern by adding a PostgreSQL trigram index on the title column through migrations to optimize ILIKE queries and update the PR. 
I also like to hear the maintainer's perspective on this approach.